app/com/arpnetworking/metrics/portal/reports/impl/EmailSender.java

```diff
@@ -32,12 +50,84 @@
 public class EmailSender implements Sender {
     @Override
     public CompletionStage<Void> send(
+            final Report report,
             final Recipient recipient,
-            final ImmutableMap<ReportFormat, RenderedReport> formatsToSend
+            final ImmutableMap<ReportFormat, RenderedReport> formatsToSend,
+            final Instant scheduled
     ) {
-        throw new RuntimeException();
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                sendSync(report, recipient, formatsToSend, scheduled);
+            } catch (final IOException e) {
+                throw new CompletionException(e);
+            }
+            return null;
+        });
     }
 
-    EmailSender() {}
+    private String sendSync(
+            final Report report,
+            final Recipient recipient,
+            final ImmutableMap<ReportFormat, RenderedReport> formatsToSend,
+            final Instant scheduled
+    ) throws IOException {
+
+        final String subject = getSubject(report, scheduled);
+        EmailPopulatingBuilder builder = EmailBuilder.startingBlank()
+                .from("no-reply+amp-reporting@dropbox.com")
+                .to(recipient.getAddress())
+                .withSubject(subject);
+
+        for (final RenderedReport rendered : formatsToSend.values()) {
+            builder = addFormat(builder, rendered);
+        }
+
+        LOGGER.info()
+                .setMessage("sending email")
+                .addData("report", report)
+                .addData("recipient", recipient)
+                .log();
+
+        _mailer.sendMail(builder.buildEmail());
+        return "";
+    }
+
+    private String getSubject(final Report report, final Instant scheduled) {
+        final String formattedTime = ZonedDateTime.ofInstant(scheduled, ZoneId.of("UTC")).toString();
+        return "[Report] " + report.getName() + " for " + formattedTime;
+    }
+
+    private EmailPopulatingBuilder addFormat(
+            final EmailPopulatingBuilder builder,
+            final RenderedReport rendered
+    ) throws IOException {
+        final String mimeType = rendered.getFormat().getMimeType();
+        final InputStream content = rendered.getBytes();
+        if (mimeType.equals("text/html")) {
+            return builder.withHTMLText(readString(content));
+        }
+        return builder.withAttachment("report", readBytes(content), mimeType);
+    }
+
+    /**
+     * Public constructor.
+     *
+     * @param mailer The Mailer to send emails through.
+     */
+    @Inject
+    public EmailSender(final Mailer mailer) {
+        _mailer = mailer;
+    }
+
+    private static byte[] readBytes(final InputStream stream) throws IOException {
+        return ByteStreams.toByteArray(stream);
```

Comment:
(this applies to `readString` as well) I'm not sure if the stream below needs to be closed but this will *not* close it https://stackoverflow.com/a/4186462